src/PanelContentsWithSubpanel/PanelContentsWithSubpanel.tsx

```diff
@@ -1,64 +0,0 @@
-import { ReactNode, useEffect, useRef, useState } from "react";
-import panelContentsWrapperStyles from "./PanelContentsWithSubpanel.module.scss";
-import { ButtonXTopRight } from "../couldBeSharedComponents/buttons/ButtonXTopRight";
-import { useCloseSubpanelWhenParentPanelCloses } from "./useCloseSubpanelWhenParentPanelCloses";
-
-export const PanelContentsWithSubpanel = ({
-  isLeftPanelOpen,
-  isSubpanelOpen = false,
-  mainPanelContent,
-  setIsSubpanelOpen = () => {},
-  subpanelClassName = undefined,
-  subpanelContent,
-}: {
-  isLeftPanelOpen: boolean;
-  isSubpanelOpen?: boolean;
-  mainPanelContent: ReactNode;
-  setIsSubpanelOpen?: (arg: boolean) => unknown;
-  subpanelClassName?: string;
-  subpanelContent: ReactNode;
-}) => {
-  const [subpanelWidthFromCss, setSubpanelWidthFromCss] = useState<string>();
-
-  const subpanelRef = useRef<HTMLDivElement>(null);
-
-  const closeSubpanel = () => {
-    setIsSubpanelOpen(false);
-  };
-
-  useCloseSubpanelWhenParentPanelCloses({
-    isLeftPanelOpen,
-    setIsSubpanelOpen,
```

Comment:
This is an over abstraction. In the new version have the internal logic in this component.